syntax/text/parser/src/Luna/Syntax/Text/Pretty/Pretty.hs

```diff
@@ -271,13 +271,26 @@ instance ( MonadIO m -- DEBUG ONLY
                                                where go imps defs = let glue = ""
                                                                     in  imps <> glue <> foldl (</>) mempty defs
 
-        AccSection   n              -> return . named (notSpaced accName) . atom $ "." <> intercalate "." (convert <$> n)
-        Metadata     t              -> return . unnamed . atom $ "###" <+> metadataHeader <+> convertVia @Text t
-        Documented   d a            -> unnamed . atom . (\a' -> "##" <> convertVia @P.String d </> a') <$> subgenBody a -- FIXME: Conversion via string is not efficient
-        Disabled     a              -> unnamed . atom . ("#" <>) <$> subgenBody a
-        Update       a ns v         -> named (spaced updateName) . atom .: (\a' v' -> convert a' <> "." <> intercalate "." (convert <$> ns) <+>              "=" <+> convert v') <$> subgen a <*> subgen v
-        Modify       a ns n v       -> named (spaced updateName) . atom .: (\a' v' -> convert a' <> "." <> intercalate "." (convert <$> ns) <+> convert n <> "=" <+> convert v') <$> subgen a <*> subgen v
-        x                           -> error $ "Pretty printer: unexpected: " <> show x <> " (" <> show root <> ")"
+        AccSection n -> return . named (notSpaced accName) . atom
+            $ "." <> intercalate "." (convert <$> n)
+        Metadata t -> return . unnamed . atom
+            $ "###" <+> metadataHeader <+> convertVia @Text t
+        -- FIXME [Ara, WD] Conversion via Text is not efficient
+        Documented d a -> unnamed . atom .
+            (\a' -> convertVia @P.String docJoined </> a') <$> subgenBody a
+            where docLines    = Text.split (== '\n') $ convertVia @P.String d
+                  docComments = map ("#" `Text.append`) docLines
+                  docJoined   = Text.intercalate "\n" docComments
+        Disabled a -> unnamed . atom . ("##" <>) <$> subgenBody a
+        Update a ns v -> named (spaced updateName) . atom .:
+            (\a' v' -> convert a' <> "." <> intercalate "." (convert <$> ns)
+            <+> "=" <+> convert v') <$> subgen a <*> subgen v
+        Modify a ns n v -> named (spaced updateName) . atom .:
+            (\a' v' -> convert a' <> "." <> intercalate "." (convert <$> ns)
+                <+> convert n <> "=" <+> convert v')
+            <$> subgen a <*> subgen v
+        x -> error $ msg <> show x <> " (" <> show root <> ")" where
+            msg = "Pretty printer: unexpected"
```

Comment:
Im not sure its correct. Does it print a space between "unexpected" and the show result?